src/libcore/num/mod.rs

```diff
@@ -2363,7 +2367,9 @@ macro_rules! uint_impl {
         /// ```
         #[stable(feature = "rust1", since = "1.0.0")]
         pub fn checked_next_power_of_two(self) -> Option<Self> {
-            let npot = self.next_power_of_two();
+            let bits = size_of::<Self>() * 8;
+            let one: Self = 1;
+            let npot = one << ((bits - self.wrapping_sub(one).leading_zeros() as usize) % bits);
```

Comment:
There are tests for `next_power_of_two` and `checked_next_power_of_two` https://github.com/rust-lang/rust/blob/master/src/libstd/num.rs#L153